packages/rum/test/rum.spec.ts

```diff
@@ -520,3 +521,96 @@ describe('rum user action', () => {
+  let lifeCycle: LifeCycle
+  let RUM: RumApi
+  let server: sinon.SinonFakeServer
+  const browserWindow = window as BrowserWindow
+  let stubBuilder: PerformanceObserverStubBuilder
+  let original: PerformanceObserver | undefined
+
+  interface ExpectedRequestBody {
+    evt: {
+      category: string
+    }
+    view: {
+      measures: {
+        first_contentful_paint: number
+      }
+    }
+  }
+
+  beforeEach(() => {
+    if (isIE()) {
+      pending('no full rum support')
+    }
+    server = sinon.fakeServer.create()
+    original = browserWindow.PerformanceObserver
+    stubBuilder = new PerformanceObserverStubBuilder()
+    browserWindow.PerformanceObserver = stubBuilder.getStub()
+    lifeCycle = new LifeCycle()
+  })
+
+  afterEach(() => {
+    browserWindow.PerformanceObserver = original
+    restorePageVisibility()
+    server.restore()
+  })
+
+  it('should not be collected when page starts not visible', () => {
+    const session = {
+      getId: () => undefined,
+      isTracked: () => true,
+      isTrackedWithResource: () => true,
+    }
+    const FAKE_RESOURCE: Partial<PerformanceEntry> = { name: 'http://foo.com', entryType: 'resource' }
+    const performanceEntry = { entryType: 'view', name: 'first-contentful-paint' }
+
+    setPageVisibility('hidden')
+    RUM = startRum('appId', lifeCycle, configuration as Configuration, session, internalMonitoring) as RumApi
+    startViewCollection(location, lifeCycle, session)
+    startPerformanceCollection(lifeCycle, session)
+    server.requests = []
+
+    stubBuilder.fakeEntry(FAKE_RESOURCE as PerformanceEntry, 'resource')
+    lifeCycle.notify(LifeCycleEventType.PERFORMANCE_ENTRY_COLLECTED, performanceEntry as PerformanceEntry)
+
+    const initialRequests = getServerRequestBodies<ExpectedRequestBody>(server)
+    expect(initialRequests.length).toBeGreaterThan(0)
+
+    initialRequests.map((request) => {
+      if (request.evt.category === 'view') {
+        expect(request.view.measures.first_contentful_paint).toBeUndefined()
+      }
+    })
+  })
+
+  it('should be collected when page starts visible', () => {
+    const session = {
+      getId: () => undefined,
+      isTracked: () => true,
+      isTrackedWithResource: () => true,
+    }
+    const FAKE_RESOURCE: Partial<PerformanceEntry> = { name: 'http://foo.com', entryType: 'resource' }
+    const performanceEntry = { entryType: 'view', name: 'first-contentful-paint' }
+
+    setPageVisibility('visible')
+    RUM = startRum('appId', lifeCycle, configuration as Configuration, session, internalMonitoring) as RumApi
+    startViewCollection(location, lifeCycle, session)
+    startPerformanceCollection(lifeCycle, session)
+    server.requests = []
+
+    stubBuilder.fakeEntry(FAKE_RESOURCE as PerformanceEntry, 'resource')
+    lifeCycle.notify(LifeCycleEventType.PERFORMANCE_ENTRY_COLLECTED, performanceEntry as PerformanceEntry)
+
+    const initialRequests = getServerRequestBodies<ExpectedRequestBody>(server)
```

Comment:
Could some of this setup be refactored with the other test case of the suite?

Also, this is a lot of setup, and maybe we could have tests a bit more unitary (I'm opening a subject here, this is more like something to think about for future development)